GDJS/Runtime/variable.ts

```diff
@@ -211,73 +285,205 @@ namespace gdjs {
      * @param newValue The new string to be set
      */
     setString(newValue: string): void {
-      this._str = newValue;
-      this._numberDirty = true;
-      this._stringDirty = false;
+      this._type = 'string';
+      this._str = '' + newValue;
+    }
+
+    /**
+     * Get the value of the variable, considered as a boolean
+     * @return {boolean}
+     */
+    getAsBoolean(): boolean {
+      if (this._type !== 'boolean') {
+        if (this._type === 'number') return this._value !== 0;
+        else if (this._type === 'string')
+          return this._str !== '0' && this._str !== '' && this._str !== 'false';
+        else return this.getChildrenCount() !== 0;
```

Comment:
Careful about this kind of implicit behavior, our choices here are very important and should be seen as such in the code (i.e: with a comment showing that an important choice was made): `// We return blabla because an object or array is considered as blabla. This aligns with blabla and is useful when checking blbla`.

In JavaScript, an object/array is always true, even when it's empty. This is because they represent something empty, but the emptiness of a collection is different from a collection not existing at all (which is represented by `null`/`undefined` in JS).

We don't have `null`/`undefined` in GDevelop - and potentially we might want to never have it (because of philosophical questions about it, and `Optional<>` being an alternative in some programming languages that is safer to use).
In any case, I would still advise to follow the JS way of doing things here. Let it be true whatever the number of children?